D107-D109 LSCOND instrumentation (units 1170-1173): call-boundary (ffc_ls_bnd_), post-main-loop (ffc_ls_mid_) and size/optical-thickness-tail (ffc_ls_tail_) record hooks in LSCOND plus ffls_pack_state (inserted after 'end subroutine LSCOND'). Generated by diff -u against the PRISTINE CLOUDS2.F90 (2026-10-05); requires ATM_DRV_clouds_lscond.f.patch (helpers ffls_*); independent of the other CLOUDS2 patches (local hunks).

--- CLOUDS2.F90.orig
+++ CLOUDS2.F90
@@ -3430,6 +3430,7 @@
     integer K,L,N  !@var K,L,N loop variables
 
     real*8 THBAR !@var THBAR potential temperature at layer edge
+    integer :: ffls_b, ffls_t   ! full-fidelity instrumentation (D107-D109)
 
     !****
     !**** LARGE-SCALE CLOUDS AND PRECIPITATION
@@ -3459,6 +3460,48 @@
     SCDNCI=SNdI
 !#endif
 
+    !**** FFLS (D107-D109) instrumentation: LSCOND call boundary, entry
+    call ffls_want(1,i_debug,j_debug,ffls_b)
+    call ffls_want(2,i_debug,j_debug,ffls_t)
+    if (ffls_b.eq.1) then
+      call ffls_put1(BYBR); call ffls_put1(RIMAX); call ffls_put1(RWMAX)
+      call ffls_put1(RWCLDOX); call ffls_put1(RCLDLX); call ffls_put1(RCLDIX)
+      call ffls_put1(WMUI); call ffls_put1(CMX); call ffls_put1(U00a)
+      call ffls_put1(U00b); call ffls_put1(RTEMP)
+      if (use_vmp) then
+        call ffls_put1(1d0)
+      else
+        call ffls_put1(0d0)
+      end if
+      call ffls_put1(dble(do_blU00)); call ffls_put1(WCONST)
+      call ffls_put1(SCDNCW); call ffls_put1(SCDNCI); call ffls_put1(BYDTsrc)
+      call ffls_put1(DTsrc)
+      call ffls_put1(PEARTH); call ffls_put1(dble(DCL))
+      call ffls_put1(dble(LMCLD)); call ffls_put1(dble(KMAX))
+    call ffls_put(QCLL,lm)
+    call ffls_put(QCIL,lm)
+    call ffls_put(SVLATL,lm)
+    call ffls_put(SVLAT1,lm)
+    call ffls_put(SVWMXL,lm)
+    call ffls_put(SDL,lm)
+    call ffls_put(VSUBL,lm)
+    call ffls_put(FSSL,lm)
+    call ffls_put(TTOLDL,lm)
+    call ffls_put(AQ,lm)
+    call ffls_put(DPDT,lm)
+    call ffls_put(PL,lm)
+    call ffls_put(PLK,lm)
+    call ffls_put(AIRM,lm)
+    call ffls_put(BYAM,lm)
+    call ffls_put(U00L,lm)
+    call ffls_put(PDSIGL00,lm)
+    call ffls_put(TAUMCL,lm)
+    call ffls_put(RNDSSL,3*lm)
+    call ffls_put(PRECNVL,lm+1)
+    call ffls_put(RA,4)
+    call ffls_pack_state
+    call ffls_stash(1)
+    end if
     !**** initialise vertical arrays
     ER=0.
     EC=0.
@@ -4634,6 +4677,23 @@
     end do CLOUD_FORMATION ! end of loop over L
 
     PRCPSS=max(0d0,PREBAR(1)*GRAV*DTsrc) ! fix small round off err
+    !**** FFLS (D107-D109): checkpoint after the main L loop, before CTEI
+    if (ffls_b.eq.1) then
+      call ffls_put1(PRCPSS); call ffls_put1(HCNDSS)
+      call ffls_pack_state
+    call ffls_put(CLEARA,lm)
+    call ffls_put(RHF,lm)
+    call ffls_put(RH00,lm)
+    call ffls_put(ER,lm)
+    call ffls_put(EC,lm)
+    call ffls_put(PREP,lm)
+    call ffls_put(QHEATL,lm)
+    call ffls_put(QHEATI,lm)
+    call ffls_put(QHEAT,lm)
+    call ffls_put(PREBAR,lm+1)
+    call ffls_put(PREICE,lm+1)
+      call ffls_flush(2,0)
+    end if
 #ifdef TRACERS_WATER
     do n=1,ntx
       TRPRSS(n)=TRPRBAR(n,1)
@@ -4924,6 +4984,44 @@
     end if ! SCMopt%allowCTEI
 #endif
 
+    !**** FFLS (D107-D109): entry of the particle size / optical thickness block
+    if (ffls_t.eq.1) then
+      call ffls_put1(PEARTH); call ffls_put1(dble(LMCLD)); call ffls_put1(dble(DCL))
+      call ffls_put1(CKIJ); call ffls_put1(HCNDSS); call ffls_put1(BYBR)
+      call ffls_put1(RIMAX); call ffls_put1(RWMAX); call ffls_put1(RWCLDOX)
+      call ffls_put1(RCLDLX); call ffls_put1(RCLDIX)
+      if (use_vmp) then
+        call ffls_put1(1d0)
+      else
+        call ffls_put1(0d0)
+      end if
+    call ffls_put(CLDSSL,lm)
+    call ffls_put(QCLX,lm)
+    call ffls_put(QCIX,lm)
+    call ffls_put(SVLHXL,lm)
+    call ffls_put(TL,lm)
+    call ffls_put(PL,lm)
+    call ffls_put(AIRM,lm)
+    call ffls_put(FSSL,lm)
+    call ffls_put(CLEARA,lm)
+    call ffls_put(QHEATL,lm)
+    call ffls_put(EC,lm)
+    call ffls_put(ER,lm)
+    call ffls_put(PREP,lm)
+    call ffls_put(WMPR,lm)
+    call ffls_put(TAUMCL,lm)
+    call ffls_put(TAUSSL,lm)
+    call ffls_put(TAUSSLIP,lm)
+    call ffls_put(CSIZEL,lm)
+    call ffls_put(CSIZELIP,lm)
+    call ffls_put(CLDSAL,lm)
+    call ffls_put(CLDSV1,lm)
+    call ffls_put(QLss,lm)
+    call ffls_put(QIss,lm)
+      call ffls_put(LHP,lm+1)
+      call ffls_stash(2)
+    end if
+
     !**** COMPUTE CLOUD PARTICLE SIZE AND OPTICAL THICKNESS
 
 #ifndef CLD_AER_CDNC
@@ -5282,9 +5380,76 @@
     end do
 #endif
 
+    !**** FFLS (D107-D109): tail exit and LSCOND call exit
+    if (ffls_t.eq.1) then
+      call ffls_put1(WMSUM)
+    call ffls_put(SVLHXL,lm)
+    call ffls_put(QCLX,lm)
+    call ffls_put(QCIX,lm)
+    call ffls_put(CLDSSL,lm)
+    call ffls_put(TAUSSL,lm)
+    call ffls_put(TAUSSLIP,lm)
+    call ffls_put(CSIZEL,lm)
+    call ffls_put(CSIZELIP,lm)
+    call ffls_put(CLDSAL,lm)
+    call ffls_put(CLDSV1,lm)
+    call ffls_put(QLss,lm)
+    call ffls_put(QIss,lm)
+    call ffls_put(WMPR,lm)
+      call ffls_flush(3,2)
+    end if
+    if (ffls_b.eq.1) then
+      call ffls_pack_state
+      call ffls_put1(PRCPSS); call ffls_put1(HCNDSS); call ffls_put1(WMSUM)
+      call ffls_put1(dble(IERR)); call ffls_put1(dble(LERR)); call ffls_put1(WMERR)
+      call ffls_flush(1,1)
+    end if
+
     return
   end subroutine LSCOND
 
+  subroutine ffls_pack_state
+!@sum full-fidelity instrumentation (D107-D109): append the LSCOND
+!@+   module-level state arrays to the current dump buffer
+!@+   (24 LM arrays, LHP, prebar1 (LM+1), QMOM, SMOM (NMOM*LM),
+!@+   UM, VM (first 4 K of each layer))
+    integer :: l
+    call ffls_put(TL,lm)
+    call ffls_put(QL,lm)
+    call ffls_put(TH,lm)
+    call ffls_put(RH,lm)
+    call ffls_put(QCLX,lm)
+    call ffls_put(QCIX,lm)
+    call ffls_put(SVLHXL,lm)
+    call ffls_put(CLDSAVL,lm)
+    call ffls_put(CLDSSL,lm)
+    call ffls_put(TAUSSL,lm)
+    call ffls_put(TAUSSLIP,lm)
+    call ffls_put(CSIZEL,lm)
+    call ffls_put(CSIZELIP,lm)
+    call ffls_put(CLDSAL,lm)
+    call ffls_put(CLDSV1,lm)
+    call ffls_put(DQLSC,lm)
+    call ffls_put(SM,lm)
+    call ffls_put(QM,lm)
+    call ffls_put(RH1,lm)
+    call ffls_put(WMPR,lm)
+    call ffls_put(QLss,lm)
+    call ffls_put(QIss,lm)
+    call ffls_put(SSHR,lm)
+    call ffls_put(DCTEI,lm)
+    call ffls_put(LHP,lm+1)
+    call ffls_put(prebar1,lm+1)
+    call ffls_put(QMOM,nmom*lm)
+    call ffls_put(SMOM,nmom*lm)
+    do l=1,lm
+      call ffls_put(UM(1,l),4)
+    end do
+    do l=1,lm
+      call ffls_put(VM(1,l),4)
+    end do
+  end subroutine ffls_pack_state
+
   !**************************************************************************************
 
   subroutine ANVIL_OPTICAL_THICKNESS(SVLATL,                         & ! input
